### PR TITLE
[flutter_tools] cleanup to devfs Operations

### LINKED ISSUE #55912: Refactor DevFSOperations class to remove references to JSON RPC write file
This is no longer used, and pending the vm service removal should be refactored so that we don't need to throw UnsupportedErrors on invocation.

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fixes https://github.com/flutter/flutter/issues/55912

removes globals from devFS operations and cleans up testing to be context free. Removes unused and unnecessarily temp directory.

### CODE REVIEW

File: packages/flutter_tools/lib/src/devfs.dart [hunk 1]
```diff
@@ -52,9 +52,6 @@ abstract class DevFSContent {
   ) {
     return osUtils.gzipLevel1Stream(contentsAsStream());
   }
-
-  /// Return the list of files this content depends on.
-  List<String> get fileDependencies => <String>[];
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/devfs.dart:57]: Unused

File: packages/flutter_tools/lib/src/devfs.dart [hunk 2]
```diff
@@ -211,60 +202,32 @@ class DevFSStringContent extends DevFSByteContent {
...
 class DevFSException implements Exception {
   DevFSException(this.message, [this.error, this.stackTrace]);
   final String message;
   final dynamic error;
   final StackTrace stackTrace;
 }
 
-class _DevFSHttpWriter {
-  _DevFSHttpWriter(
+class DevFSHttpWriter {
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/devfs.dart:212]: Can this be `@visibleForTesting`?

**Review by zanderso — APPROVED** [review]
lgtm w/ question

File: packages/flutter_tools/lib/src/devfs.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/devfs.dart:212]: I think it could actually be made private again

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/devfs.dart]: Done

**Pull request merged**